mono/io-layer/wthreads.c

```diff
@@ -358,7 +352,11 @@ GetThreadPriority (gpointer handle)
 				  (gpointer *)&thread_handle);
 				  
 	if (ok == FALSE) {
-		return (THREAD_PRIORITY_NORMAL);
+		pthread_getschedparam (pthread_self(), &policy, &param);
+		if ((policy == SCHED_FIFO) || (policy == SCHED_RR)) 
+			return (((MonoInternalThread *)handle)->priority);
```

Comment:
GetThreadPriority() is only call by way of:
```
int
ves_icall_System_Threading_Thread_GetPriority (MonoThread *this_obj)
{
        gint32 priority;
        MonoInternalThread *internal = this_obj->internal_thread;

        LOCK_THREAD (internal);
        priority = GetThreadPriority (internal) + 2;
        UNLOCK_THREAD (internal);
        return priority;
}
```
So the handle is (in reality) a pointer to MonoInternalThread. I'm not sure how else to access the priority prior to thread start.